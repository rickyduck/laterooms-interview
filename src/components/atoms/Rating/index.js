import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { IconButton } from '@material-ui/core';
import orange from '@material-ui/core/colors/orange';
import grey from '@material-ui/core/colors/grey';
import Star from '@material-ui/icons/Star';
import StarBorder from '@material-ui/icons/StarBorder';

const styles = {
  disabled: {
    pointerEvents: 'none'
  }
};

export default class Rating extends Component {
  constructor(props) {
    super(props);
  }

  renderIcon(i) {
    const filled = i <= this.props.value;

    if (filled) {
      return this.props.iconFilledRenderer
        ? this.props.iconFilledRenderer({
            ...this.props,
            index: i
          })
        : this.props.iconFilled;
    } else {
      return this.props.iconNormalRenderer
        ? this.props.iconNormalRenderer({
            ...this.props,
            index: i
          })
        : this.props.iconNormal;
    }
  }

  render() {
    const {
      disabled,
      iconFilled,
      iconHovered,
      iconNormal,
      iconFilledRenderer,
      iconHoveredRenderer,
      iconNormalRenderer,
      itemStyle,
      itemClassName,
      itemIconStyle,
      max,
      style,
      value,
      ...other
    } = this.props;

    const rating = [];

    for (let i = 1; i <= max; i++) {
      rating.push(
        <IconButton
          key={i}
          className={itemClassName}
          iconStyle={itemIconStyle}
          style={itemStyle}
        >
          {this.renderIcon(i)}
        </IconButton>
      );
    }

    return (
      <div
        style={
          this.props.disabled || this.props.readOnly
            ? { ...styles.disabled, ...this.props.style }
            : this.props.style
        }
        {...other}
      >
        {rating}
      </div>
    );
  }
}
Rating.defaultProps = {
  disabled: false,
  iconFilled: <Star className="filled" fill={orange['500']} />,
  iconNormal: <StarBorder className="empty" fill={orange['300']} />,
  max: 5,
  readOnly: false,
  value: 0
};

Rating.propTypes = {
  iconFilled: PropTypes.node,
  iconNormal: PropTypes.node,
  max: PropTypes.number,
  style: PropTypes.object,
  value: PropTypes.number
};
