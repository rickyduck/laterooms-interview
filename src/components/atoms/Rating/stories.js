import React from 'react';
import { withKnobs, number } from '@storybook/addon-knobs';

import { storiesOf } from '@storybook/react';
import Rating from './';

const stories = storiesOf('Atoms', module);
stories.addDecorator(withKnobs);

stories.add('Rating', () => (
  <Rating value={number('Value', 3)} max={number('Max', 5)} />
));
