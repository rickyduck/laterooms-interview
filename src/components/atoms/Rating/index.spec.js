import React from 'react';
import { render } from 'react-testing-library';

import Rating from './';

test('Renders Rating correctly', () => {
  const { container } = render(<Rating max={5} value={1} />);
  expect(container).toMatchSnapshot();
});
