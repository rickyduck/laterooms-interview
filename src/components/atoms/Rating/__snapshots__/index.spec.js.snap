// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Rating correctly 1`] = `
<div>
  <div>
    <button
      class="MuiButtonBase-root-7 MuiIconButton-root-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-6"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-10 filled"
          fill="#ff9800"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
          <path
            d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-19"
      />
    </button>
    <button
      class="MuiButtonBase-root-7 MuiIconButton-root-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-6"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-10 empty"
          fill="#ffb74d"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-19"
      />
    </button>
    <button
      class="MuiButtonBase-root-7 MuiIconButton-root-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-6"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-10 empty"
          fill="#ffb74d"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-19"
      />
    </button>
    <button
      class="MuiButtonBase-root-7 MuiIconButton-root-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-6"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-10 empty"
          fill="#ffb74d"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-19"
      />
    </button>
    <button
      class="MuiButtonBase-root-7 MuiIconButton-root-1"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiIconButton-label-6"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root-10 empty"
          fill="#ffb74d"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </span>
      <span
        class="MuiTouchRipple-root-19"
      />
    </button>
  </div>
</div>
`;
